src/utilities/createRouter.ts

```diff
@@ -36,37 +45,36 @@ export function createRouter<const T extends Routes>(routes: T, options: RouterO
   } = createRouterHooks()
 
   async function update(url: string, { replace }: RouterUpdateOptions = {}): Promise<void> {
+    history.stopListening()
+
     const to = getResolvedRouteForUrl(routes, url) ?? getRejectionRoute('NotFound')
     const from = route
 
     const beforeResponse = await runBeforeRouteHooks({ to, from, hooks })
 
-    const setHistory = (): void => history.update(url, { replace })
-    const setRoute = (): void => updateRoute(to)
-
     switch (beforeResponse.status) {
       // On abort do nothing
       case 'ABORT':
         return
 
       // On push update the history, and push new route, and return
       case 'PUSH':
-        setHistory()
-        push(...beforeResponse.to)
+        history.update(url, { replace })
+        await push(...beforeResponse.to)
```

Comment:
This makes sense but explain why cause I'm not sure it makes a difference

Reply:
The await? I just wanna make sure we don't end up in a thread that starts listening, then we come back to this thread and it was left on.